[Release] Add History Mode Functionality to Salesforce Source Package (#40)
fivetran/dbt_salesforce_source: README.md

    search_order: ['spark_utils', 'dbt_utils']
```

### Database Incremental Strategies 

> Some of the end models in this package are materialized incrementally. We have chosen `insert_overwrite` as the default strategy for **BigQuery** and **Databricks** databases, as it is only available for these dbt adapters. For **Snowflake**, **Redshift**, and **Postgres** databases, we have chosen `delete+insert` as the default strategy.

Reviewer: ```suggestion
The history models in this package are materialized incrementally. We have chosen `insert_overwrite` as the default strategy for **BigQuery** and **Databricks** databases, as it is only available for these dbt adapters. For **Snowflake**, **Redshift**, and **Postgres** databases, we have chosen `delete+insert` as the default strategy.
```

Author: Updated!

## Step 2: Installing the Package (skip if also using the `salesforce` transformation package)
If you are **not** using the [Salesforce transformation package](https://github.com/fivetran/dbt_salesforce), include the following `salesforce_source` package version in your `packages.yml`
> Check [dbt Hub](https://hub.getdbt.com/) for the latest installation instructions, or [read the dbt docs](https://docs.getdbt.com/docs/package-management) for more information on installing packages.

```yaml
packages:
  - package: fivetran/salesforce_source
    version: [">=1.0.0", "<1.1.0"] # we recommend using ranges to capture non-breaking changes automatically
```
## Step 3: Configure Your Variables
### Database and Schema Variables (Using the standard Salesforce schema only)
By default, this package will run using your target database and the `salesforce` schema. If this is not where your Salesforce data is (perhaps your Salesforce schema is `salesforce_fivetran`), add the following configuration to your root `dbt_project.yml` file:

```yml
vars:
    salesforce_database: your_database_name
    salesforce_schema: your_schema_name
```
 

> See the below section for further configurations if you are leveraging Salesforce History Mode.

Author: Removed.

```
The corresponding metrics from the disabled tables will not populate in the downstream models.

> ## Optional: Utilizing Salesforce History Mode records

Reviewer: This is important enough to be it's own step. I would number it and then adjust the downstream numbers to account for the new section.

```suggestion
## (Optional) Step 4: Utilizing Salesforce History Mode records
```

Author: Steps updated!

### IMPORTANT: How To Update Your History Models
To ensure maximum value for these history mode models and avoid messy historical data that could come with picking and choosing which fields you bring in, **all fields in your Salesforce history mode connector are being synced into your end staging models**. That means all custom fields you picked to sync are being brought in to the final models. [See our DECISIONLOG for more details on why we are bringing in all fields](https://github.com/fivetran/dbt_salesforce_source/blob/main/DECISIONLOG.md). 

To update the history mode models, you must follow these steps: 
1) Go to your Fivetran Salesforce History Mode connector page.
2) Update the fields that you are bringing into the model. 
3) Run a `dbt run --full-refresh` on the specific staging models you've updated to bring in these fields and all the historical data available with these fields.

> We are aware that bringing in additional fields will be very process-heavy, so we do emphasize caution in making changes to your history mode connector. It would be best to batch as many field changes as possible before executing a `--full-refresh` to save on processing. 

Author: Moved.

      alias: "user_role_field_x"
  salesforce__user_pass_through_columns: 
    - name: "salesforce__user_field"

  ##History mode passthrough columns
  salesforce__account_history_pass_through_columns:
    - name: "salesforce__account_history_field"
      alias: "account_history_field_x"
  salesforce__contact__history_pass_through_columns:
    - name: "salesforce__contact_history_field"
      alias: "contact_history_field_x"
  salesforce__event__history_pass_through_columns:
    - name: "salesforce__event_history_field"
      alias: "event_history_field_x"
  salesforce__lead__history_pass_through_columns:
    - name: "salesforce__lead_history_field"

>       alias: "lead_history_field_x"

Author: I think they can all be removed since we are no longer utilizing passthrough columns on any of these history models?

  salesforce__opportunity_history_pass_through_columns:
    - name: "salesforce__opportunity_history_field"
      alias: "opportunity_history_field_x"
  salesforce__opportunity_line_item_history__pass_through_columns:
    - name: "salesforce__opportunity_line_item_history_field"
      alias: "opportunity_line_item_history_field_x"
  salesforce__product_2_history_pass_through_columns:
    - name: "salesforce__product_2_history_field"
      alias: "product_2_history_field_x"  
  salesforce__task_history_pass_through_columns:
    - name: "salesforce__task_history_field"
      alias: "task_history_field_x"
  salesforce__user_history_pass_through_columns:
    - name: "salesforce__user_history_field"
      alias: "user_history_field_x"
  salesforce__user_role_history_pass_through_columns:
    - name: "salesforce__user_role_history_field"

>       alias: "user_role_history_field_x"

Author: Removed!
